api/v1/server/server.go

```
@@ -180,24 +188,182 @@ func (s *Server) Stop() {
 	close(s.quit)
 }
 
+func (s *Server) Login(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	un, pw, ok := r.BasicAuth()
+	if !ok {
+		w.Header().Set("WWW-Authenticate", `Basic realm="Enter Login for API Token"`)
+		http.Error(w, "Not Authorized", http.StatusUnauthorized)
+		return
+	}
+	tok, err := s.auth.LoginAndTokenize(un, pw)
+	if err != nil {
+		log.Error(err)
+		w.Header().Set("WWW-Authenticate", `Basic realm="Enter Login for API Token"`)
+		http.Error(w, "Bad Login", http.StatusUnauthorized)
+		return
+	}
+	apiTok := convert.FromAuthToken(tok)
+	w.WriteHeader(http.StatusOK)
+	if err = json.NewEncoder(w).Encode(apiTok); err != nil {
+		log.Errorf("Unable to Encode Token: %v", err)
```

do we want to return an http error here? bad request?